tests/draft2019-09/prototype-safe.json

```diff
@@ -0,0 +1,624 @@
+        "tests": [
+            { "description": "Valid on numbers", "data": 0, "valid": true },
+            { "description": "Valid on arrays", "data": [], "valid": true },
+            { "description": "Valid if not present", "data": {}, "valid": true },
+            { "description": "Valid if correct", "data": { "x": {} }, "valid": true },
+            { "description": "Invalid if incorrect (string)", "data": { "x": "foo" }, "valid": false },
+            { "description": "Invalid if incorrect (empty string)", "data": { "x": "" }, "valid": false },
+            { "description": "Invalid if incorrect (boolean true)", "data": { "x": true }, "valid": false },
+            { "description": "Invalid if incorrect (boolean false)", "data": { "x": false }, "valid": false },
+            { "description": "Invalid if incorrect (array)", "data": { "x": [42] }, "valid": false },
+            { "description": "Invalid if incorrect (empty array)", "data": { "x": [] }, "valid": false },
+            { "description": "Invalid if incorrect (zero number)", "data": { "x": 0 }, "valid": false },
+            { "description": "Invalid if incorrect (number)", "data": { "x": 42 }, "valid": false }
+        ]
+    },
+    {
+        "description": "Does not see elements non existing on the object: 'x' via required",
+        "comment": "This is a baseline check to validate that other properties are treated the same and are not special compared to it",
+        "schema": {
+            "required": ["x"]
+        },
+        "tests": [
+            { "description": "Valid on numbers", "data": 0, "valid": true },
+            { "description": "Valid on arrays", "data": [], "valid": true },
+            { "description": "Invalid if not present", "data": {}, "valid": false },
+            { "description": "Valid if present (string)", "data": { "x": "foo" }, "valid": true },
+            { "description": "Valid if present (empty string)", "data": { "x": "" }, "valid": true },
+            { "description": "Valid if present (boolean true)", "data": { "x": true }, "valid": true },
+            { "description": "Valid if present (boolean false)", "data": { "x": false }, "valid": true },
+            { "description": "Valid if present (array)", "data": { "x": [42] }, "valid": true },
+            { "description": "Valid if present (empty array)", "data": { "x": [] }, "valid": true },
+            { "description": "Valid if present (object)", "data": { "x": {} }, "valid": true },
+            { "description": "Valid if present (zero number)", "data": { "x": 0 }, "valid": true },
+            { "description": "Valid if present (number)", "data": { "x": 42 }, "valid": true }
+        ]
+    },
+    {
+        "description": "Does not see elements non existing on the object: 'length' as number",
+        "comment": "This is a common bug of some of the JS validators, this test detects it",
+        "schema": {
+            "properties": { "length": { "type": "number" } }
```

Comment:
Is there any implementation that treats "length" as special?

Reply:
`length` is a property on arrays. If the implementation applies `properties` checks to objects too, it might fail.

Reply:
`djv` fails only on `length`.

I hadn't known that in advance and did not design this with `djv` in mind, but I did expect that there are multiple possible failure points.
This is why these tests need to check a variety of properties..